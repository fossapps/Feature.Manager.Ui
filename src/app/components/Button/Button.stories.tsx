import * as React from "react";
import { Color } from "../../constants";
import { withInitialState } from "../../helpers/withInitialState";
import { Button } from "./Button";

export default {
  component: Button,
  title: "Button"
};

export const Primary = () => <Button>Primary</Button>;
export const Secondary = () => <Button type={"secondary"}>Secondary</Button>;
export const Disabled = () => <Button disabled={true}>Disabled</Button>;
export const Success = () => <Button type="success">Success</Button>;
export const Error = () => <Button type="error">Error</Button>;
export const Small = () => <Button btnSize="small">Small</Button>;
export const Default = () => <Button>Default</Button>;
export const Large = () => <Button btnSize="large">Large</Button>;
export const StyledButton = () => {
  const style = {
    backgroundColor: Color.BLACK,
    border: 0,
    borderRadius: 5,
    color: Color.BLUE
  };
  return (
    <Button style={style}>
      Styled
    </Button>
  );
};
export const SwitchableButton = ({ state, setState }) => (
  <Button
    type={state.type}
    onClick={() => setState({ type: state.type === "primary" ? "secondary" : "primary" })}
  >
    Click me!
  </Button>
);
SwitchableButton.story = {
  decorators: [withInitialState({ type: "primary" })]
};
