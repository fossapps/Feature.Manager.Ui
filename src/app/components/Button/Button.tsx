import * as React from "react";
import { classes, stylesheet } from "typestyle";
import { Color } from "../../constants/Color";

const classNames = stylesheet({
  button: {
    border: "none",
    borderRadius: "5px",
    outline: "none",
    transition: ".3s",

    $nest: {
      "&:hover": {
        opacity: ".8"
      }
    }
  },
  danger: {
    backgroundColor: Color.DANGER,
    color: Color.WHITE,
    cursor: "pointer"
  },
  default: {
    fontSize: "16px",
    padding: "10px 25px"
  },
  disabled: {
    backgroundColor: Color.GREY,
    border: "none",
    color: Color.BLACK,
    cursor: "not-allowed"
  },
  large: {
    fontSize: "18px",
    padding: "20px 25px"
  },
  primary: {
    backgroundColor: Color.PRIMARY,
    color: Color.WHITE,
    cursor: "pointer"
  },
  secondary: {
    backgroundColor: Color.SECONDARY,
    color: Color.WHITE,
    cursor: "pointer"
  },
  small: {
    fontSize: "14px",
    padding: "5px 15px"
  },
  success: {
    backgroundColor: Color.SUCCESS,
    color: Color.WHITE,
    cursor: "pointer"
  }
});

export type TButtonType = "primary" | "secondary" | "danger" | "success";
export type TButtonSize = "small" | "default" | "large";

interface IProps extends React.HTMLProps<HTMLButtonElement> {
  btnSize?: TButtonSize;
  /**
   * Either primary or secondary
   */
  type?: TButtonType;
}

export class Button extends React.Component<IProps> {
  public static defaultProps: Partial<IProps> = {
    btnSize: "default",
    type: "primary"
  };

  public render(): JSX.Element {
    const {
      children,
      className,
      disabled,
      type,
      btnSize,
      ...rest
    } = this.props;
    const styles = classes(
      disabled ? classNames.disabled : classNames[type],
      !btnSize ? classNames.default : classNames[btnSize]
    );
    return (
      <button
        className={classes(classNames.button, styles, className)}
        disabled={disabled}
        {...rest}
      >
        {children}
      </button>
    );
  }
}
