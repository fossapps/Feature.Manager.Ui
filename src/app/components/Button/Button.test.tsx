import { shallow } from "enzyme";
import * as React from "react";
import { Button } from "./Button";

describe("Button", () => {
  it("should match snapshot", () => {
    expect(shallow(<Button>Default</Button>)).toMatchSnapshot();
    expect(shallow(<Button type={"danger"}>Error</Button>)).toMatchSnapshot();
    expect(shallow(<Button type={"primary"}>Primary</Button>)).toMatchSnapshot();
    expect(shallow(<Button type={"secondary"}>Secondary</Button>)).toMatchSnapshot();
    expect(shallow(<Button type={"success"}>Success</Button>)).toMatchSnapshot();
    expect(shallow(<Button btnSize={"default"}>Success</Button>)).toMatchSnapshot();
    expect(shallow(<Button btnSize={"small"}>Success</Button>)).toMatchSnapshot();
    expect(shallow(<Button btnSize={"large"}>Success</Button>)).toMatchSnapshot();
  });
});
