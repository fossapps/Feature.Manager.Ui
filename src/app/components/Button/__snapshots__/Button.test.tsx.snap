// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
<button
  className="button_fbnlpvc primary_f1bai8sv default_f1tpsmcd"
>
  Default
</button>
`;

exports[`Button should match snapshot 2`] = `
<button
  className="button_fbnlpvc error_f1f5vffv default_f1tpsmcd"
>
  Error
</button>
`;

exports[`Button should match snapshot 3`] = `
<button
  className="button_fbnlpvc primary_f1bai8sv default_f1tpsmcd"
>
  Primary
</button>
`;

exports[`Button should match snapshot 4`] = `
<button
  className="button_fbnlpvc secondary_fr1jdjh default_f1tpsmcd"
>
  Secondary
</button>
`;

exports[`Button should match snapshot 5`] = `
<button
  className="button_fbnlpvc success_f1a4qxdm default_f1tpsmcd"
>
  Success
</button>
`;

exports[`Button should match snapshot 6`] = `
<button
  className="button_fbnlpvc primary_f1bai8sv default_f1tpsmcd"
>
  Success
</button>
`;

exports[`Button should match snapshot 7`] = `
<button
  className="button_fbnlpvc primary_f1bai8sv small_ftp1rns"
>
  Success
</button>
`;

exports[`Button should match snapshot 8`] = `
<button
  className="button_fbnlpvc primary_f1bai8sv large_f1iu98hc"
>
  Success
</button>
`;
